shared/constants/network.ts

```diff
@@ -0,0 +1,675 @@
+export const PALM_DISPLAY_NAME = 'Palm';
+
+export const infuraProjectId = process.env.INFURA_PROJECT_ID;
+export const getRpcUrl = ({
+  network,
+  excludeProjectId = false,
+}: {
+  network: NetworkType;
+  excludeProjectId?: boolean;
+}) =>
+  `https://${network}.infura.io/v3/${excludeProjectId ? '' : infuraProjectId}`;
+
+export const ROPSTEN_RPC_URL = getRpcUrl({
+  network: NETWORK_TYPES.ROPSTEN,
+});
+export const RINKEBY_RPC_URL = getRpcUrl({
+  network: NETWORK_TYPES.RINKEBY,
+});
+export const KOVAN_RPC_URL = getRpcUrl({ network: NETWORK_TYPES.KOVAN });
+export const MAINNET_RPC_URL = getRpcUrl({
+  network: NETWORK_TYPES.MAINNET,
+});
+export const GOERLI_RPC_URL = getRpcUrl({ network: NETWORK_TYPES.GOERLI });
+export const LOCALHOST_RPC_URL = 'http://localhost:8545';
+
+/**
+ * An object containing the token symbols for various tokens that are either
+ * native currencies or those that have been special cased by the extension
+ * for supporting our feature set.
+ */
+export const CURRENCY_SYMBOLS = {
+  ARBITRUM: 'ETH',
+  AVALANCHE: 'AVAX',
+  BNB: 'BNB',
+  BUSD: 'BUSD',
+  CELO: 'CELO',
+  DAI: 'DAI',
+  ETH: 'ETH',
+  FANTOM: 'FTM',
+  HARMONY: 'ONE',
+  PALM: 'PALM',
+  MATIC: 'MATIC',
+  TEST_ETH: 'TESTETH',
+  USDC: 'USDC',
+  USDT: 'USDT',
+  WETH: 'WETH',
+} as const;
+
+export const ETH_TOKEN_IMAGE_URL = './images/eth_logo.svg';
+export const TEST_ETH_TOKEN_IMAGE_URL = './images/black-eth-logo.svg';
+export const BNB_TOKEN_IMAGE_URL = './images/bnb.png';
+export const MATIC_TOKEN_IMAGE_URL = './images/matic-token.png';
+export const AVAX_TOKEN_IMAGE_URL = './images/avax-token.png';
+export const AETH_TOKEN_IMAGE_URL = './images/arbitrum.svg';
+export const FTM_TOKEN_IMAGE_URL = './images/fantom-opera.svg';
+export const HARMONY_ONE_TOKEN_IMAGE_URL = './images/harmony-one.svg';
+export const OPTIMISM_TOKEN_IMAGE_URL = './images/optimism.svg';
+export const PALM_TOKEN_IMAGE_URL = './images/palm.svg';
+
+export const INFURA_PROVIDER_TYPES = [
+  NETWORK_TYPES.ROPSTEN,
+  NETWORK_TYPES.RINKEBY,
+  NETWORK_TYPES.KOVAN,
+  NETWORK_TYPES.MAINNET,
+  NETWORK_TYPES.GOERLI,
+];
+
+export const TEST_CHAINS = [
+  CHAIN_IDS.ROPSTEN,
+  CHAIN_IDS.RINKEBY,
+  CHAIN_IDS.GOERLI,
+  CHAIN_IDS.KOVAN,
+  CHAIN_IDS.LOCALHOST,
+];
+
+const typedCapitalize = <K extends string>(k: K): Capitalize<K> =>
+  capitalize(k) as Capitalize<typeof k>;
+
+export const TEST_NETWORK_TICKER_MAP: {
+  [K in Exclude<
+    NetworkType,
+    'localhost' | 'mainnet' | 'rpc'
+  >]: `${Capitalize<K>}${typeof CURRENCY_SYMBOLS.ETH}`;
+} = {
+  [NETWORK_TYPES.ROPSTEN]: `${typedCapitalize(NETWORK_TYPES.ROPSTEN)}${
+    CURRENCY_SYMBOLS.ETH
+  }`,
+  [NETWORK_TYPES.RINKEBY]: `${typedCapitalize(NETWORK_TYPES.RINKEBY)}${
+    CURRENCY_SYMBOLS.ETH
+  }`,
+  [NETWORK_TYPES.KOVAN]: `${typedCapitalize(NETWORK_TYPES.KOVAN)}${
+    CURRENCY_SYMBOLS.ETH
+  }`,
+  [NETWORK_TYPES.GOERLI]: `${typedCapitalize(NETWORK_TYPES.GOERLI)}${
+    CURRENCY_SYMBOLS.ETH
+  }`,
+} as const;
```

Comment:
How does `as const` here helps ?

Reply:
Good catch, it helped before the stronger typing to keep the types to being broadened to 'string' but with the map type now specified it is no longer needed